### PR TITLE
feat: add callcode

### PR DESCRIPTION
**Pull request by bajpai244**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

callcode opcode isn't implemented.

Resolves: #395 

## What is the new behavior?

- callcode opcode is implemented

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/evm/src/instructions/system_operations.cairo [hunk 1]
```diff
@@ -112,7 +112,11 @@ impl SystemOperations of SystemOperationsTrait {
     /// CALLCODE
     /// # Specification: https://www.evm.codes/#f2?fork=shanghai
     fn exec_callcode(ref self: Machine) -> Result<(), EVMError> {
-        Result::Err(EVMError::NotImplemented)
+        let call_args = self.prepare_call(@CallType::DelegateCall)?;
```
Comment (enitrat) [inline crates/evm/src/instructions/system_operations.cairo:115]:
```suggestion
        let call_args = self.prepare_call(@CallType::CallCode)?;
```

Comment (bajpai244) [reply crates/evm/src/instructions/system_operations.cairo:115]: Ah, nice catch!

Comment (bajpai244) [reply crates/evm/src/instructions/system_operations.cairo]: fixed via [this](https://github.com/kkrt-labs/kakarot-ssj/pull/559/commits/e2bdcf6d3dd78569a01fe906e6c4efe5fe8affdd) commit.

**Review by enitrat — APPROVED** [review]
lgtm

**Pull request merged**